core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -121,6 +122,10 @@ class SparkHadoopUtil extends Logging {
     UserGroupInformation.loginUserFromKeytab(principalName, keytabFilename)
   }
 
+  def setPrincipalAndKeytabForLogin(principal: String, keytab: String): Unit = {}
+
+  private[spark] def scheduleLoginFromKeytab(callback: (SerializableBuffer) => Unit): Unit = {}
```

Comment:
Should this be `= ???` instead of `= {}`?

Just so if it's called when it shouldn't, it complains.